pyccel/ast/headers.py

```diff
@@ -667,11 +710,70 @@ def apply(self, args, results=None):
             for i, arg in d_arguments.items():
                 if isinstance(arg, Macro):
                     d_arguments[i] = construct_macro(arg.name,
-                                      d_arguments[arg.argument])
+                                                     d_arguments[arg.argument])
                     if isinstance(arg, MacroShape):
                         d_arguments[i]._index = arg.index
+        return d_arguments
 
 
+    def get_results_shapes(self, args):
+        """replace elements of the shape with appropriate values
+
+        Parameters
+        ----------
+        args: iterable
+            The arguments provided at function call
+
+        Results
+        -------
+        results_shapes: iterable
+            List of shapes after replacing variables indicated
+            in the macro if they exist, with the appropriate variables
+            from args.
+        """
+
+        d_arguments = self.link_args(args)
+        argument_keys = d_arguments.keys()
+        results_shapes = []
+        for j, shape in enumerate(self.results_shapes):
+            newargs = []
+
+            for i, arg in enumerate(shape):
+                if str(arg) == ':':
+                    try:
+                        new = MacroShape(d_arguments[self.results[j]], i)
+                    except KeyError:
+                        msg = "Shape needs to be provided explicitly as it cannot be deduced"
+                        errors.report(msg, symbol=self.results[j],
```

Comment:
I think `zip` makes more sense here than `enumerate`. You never use `j` directly

Reply:
Fixed 18df3a1